props/blaster.h

```diff
@@ -260,9 +233,65 @@ class Blaster : public PROP_INHERIT_PREFIX PropBase {
     }
   }
 
+   RefPtr<BufferedWavPlayer> auto_player_;
+
+  uint32_t auto_time_;
+  virtual void Fire() {
+    CheckEmpty();
+    if (empty_) return;
+#ifdef ENABLE_MOTION
+#ifdef BLASTER_JAM_PERCENTAGE
+    // If we're already jammed then we don't need to recheck. If we're not jammed then check if we just jammed.
+    is_jammed_ = is_jammed_ ? true : CheckJam(BLASTER_JAM_PERCENTAGE);
+    if (is_jammed_) {
+      SaberBase::DoEffect(EFFECT_JAM, 0);
+      return;
+    }
+#endif
+#endif
+    if (blaster_mode == MODE_AUTO) {
+      // Set up the autofire pairing if the font suits it.
+      SelectAutoFirePair(); 
+      SaberBase::SetLockup(LOCKUP_AUTOFIRE);
+      SaberBase::DoBeginLockup();
+      auto_firing_ = true;
+      auto_time_ = millis();
+
+      // GET auto.wav here
+      if (!auto_player_) {
+        auto_player_ = GetWavPlayerPlaying(&SFX_auto);
+        }
+    } else {
+      if (blaster_mode == MODE_STUN) {
+        SaberBase::DoEffect(EFFECT_STUN, 0);
+        STDOUT << "******** STUN - Remaining shots = " << GetBulletCount() << "\n";
+      } else {
+        SaberBase::DoEffect(EFFECT_FIRE, 0);
+        STDOUT << "******** FIRE - Remaining shots = " << GetBulletCount() << "\n";
+      }
+      shots_fired_++;
+    }
+  }
+
   void Loop() override {
     PropBase::Loop();
     PollNextAction();
+    if (auto_firing_) {
+      if (&auto_player_) {
+        if (millis() - auto_time_ > 1000 * auto_player_->length()) {
+          shots_fired_++;
+          auto_time_ = millis();
+          STDOUT << "******** AUTOFIRING - Remaining shots = " << GetBulletCount() << "\n";
+          CheckEmpty();
+          if (empty_) {
+            SaberBase::DoEndLockup();
+            SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
+            auto_firing_ = false;
```

Comment:
We should call free on the auto_player_ here.

Reply:
Shoot. I had it there but thought you preferred it not in Loop(), so I put it on the button release. 
OK.

Reply:
It should be everywhere where we finish auto fire mode.

Reply:
yup. done.